test/acceptance/testrunners/filepermissiontestrunner.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package testrunners
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/filesystem"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/rule"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"log"
+	"time"
+)
+
+type FilePermissionTestRunner struct {
+	test_runner.BaseTestRunner
+}
+
+var _ test_runner.ITestRunner = (*FilePermissionTestRunner)(nil)
+
+const (
+	agentConfigPath      = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+	agentConfigCopiedDir = "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d"
+	translatedTomlPath   = "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml"
+)
+
+var (
+	onlyRootCanWriteRule = rule.Rule[string]{
+		Conditions: []rule.ICondition[string]{
+			&rule.PermittedEntityMatch{ExpectedOwner: aws.String("root"), ExpectedGroup: aws.String("root")},
+			&rule.FilePermissionExpected{PermissionCompared: filesystem.OwnerWrite, ShouldExist: true},
+			&rule.FilePermissionExpected{PermissionCompared: filesystem.AnyoneWrite, ShouldExist: false},
+		},
+	}
+	onlyCWAgentCanWriteRule = rule.Rule[string]{
+		Conditions: []rule.ICondition[string]{
+			&rule.PermittedEntityMatch{ExpectedOwner: aws.String("cwagent"), ExpectedGroup: aws.String("cwagent")},
+			&rule.FilePermissionExpected{PermissionCompared: filesystem.OwnerWrite, ShouldExist: true},
+			&rule.FilePermissionExpected{PermissionCompared: filesystem.AnyoneWrite, ShouldExist: false},
+		},
+	}
+)
+
+var testCases = map[string]rule.Rule[string]{
+	agentConfigPath:      onlyRootCanWriteRule,
+	agentConfigCopiedDir: onlyCWAgentCanWriteRule,
+	translatedTomlPath:   onlyRootCanWriteRule,
+}
+
+var testGroupResult *status.TestGroupResult = nil
+
+func (m *FilePermissionTestRunner) Validate() status.TestGroupResult {
```

Comment:
Shouldn't the validation be done in the function called `Validate()`?
Why is `validatePermissions` called in `SetupAfterAgentRun()`?

Reply:
Yeah what was initially not considered in the very beginning was that not all validations will happen via calling CW backend. iirc some of the files(?) we need to validate no longer existed after stopping agent.

Reply:
yeah i just encountered this issue, i'm moving it back and leaving a comment

Reply:
looks like this line is the culprit 

https://github.com/aws/amazon-cloudwatch-agent-test/blob/main/test/test_runner/base_test_runner.go#L108